3rdparty/CMakeLists.txt

```diff
@@ -222,6 +222,17 @@ unset(CMAKE_INSTALL_MANDIR)
 # blake3
 add_subdirectory(blake/c EXCLUDE_FROM_ALL)
 
+# Deflate
+set(LIBDEFLATE_BUILD_SHARED_LIB OFF)
+set(LIBDEFLATE_BUILD_STATIC_LIB ON)
+set(LIBDEFLATE_GZIP_SUPPORT OFF)
+set(LIBDEFLATE_BUILD_GZIP OFF)
+add_subdirectory(libdeflate EXCLUDE_FROM_ALL)
+set(libdeflate_DIR "${CMAKE_CURRENT_BINARY_DIR}/libdeflate")
+
+# Imath
+add_subdirectory(imath EXCLUDE_FROM_ALL)
+
 # OpenEXR
 # note we are adding it anyway regardless _NBL_COMPILE_WITH_OPEN_EXR_, its because we need Half dependecy 
 # it generates on fly in its build directory which doesnt have to be linked (class with inline methods), 
```

Comment:
since half.h is now part of IMath 3.0 we can disable OpenEXR (and Deflate) when it's not needed

Reply:
okay go ahead

Reply:
Checked libs `OpenEXRUtil`, `OpenEXRCore`, `Iex`, `IlmThread` (OpenEXR libs); Neither Nabla, nor any other dependency import their headers. I assume it's safe to make OpenEXR optional.